xline/src/utils.rs

```diff
@@ -0,0 +1,45 @@
+use parking_lot::{RwLock, RwLockReadGuard, RwLockWriteGuard};
+
+/// Apply a closure on a rwlock after getting the guard
+pub(crate) trait RwLockMap<T, R> {
```

Comment:
I think there's an util for the lock in CURP. shall we merge them?

Reply:
I will merge them to a new crate in next PR